lib/utils/bloc/auth/auth_bloc.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:hackpsu/data/authentication_repository.dart';
+import 'package:hackpsu/utils/bloc/auth/auth_event.dart';
+import 'package:hackpsu/utils/bloc/auth/auth_state.dart';
+
+class AuthBloc extends Bloc<AuthEvent, AuthState> {
+  AuthBloc({
+    @required AuthenticationRepository authenticationRepository,
+  })  : _authenticationRepository = authenticationRepository,
+        super(
+          authenticationRepository.currentUser != null
+              ? AuthState.authenticated(authenticationRepository.currentUser)
+              : AuthState.unauthenticated(),
+        ) {
+    on<AuthUserChanged>(_onAuthUserChanged);
+    on<AuthLogout>(_onAuthLogout);
+    _userSubscription = _authenticationRepository.user.listen(
+      (user) => add(AuthUserChanged(user)),
+    );
+  }
+
+  final AuthenticationRepository _authenticationRepository;
+  StreamSubscription<User> _userSubscription;
+
+  void _onAuthUserChanged(AuthUserChanged event, Emitter<AuthState> emit) {
+    emit(
+      event.user != null
+          ? AuthState.authenticated(event.user)
+          : AuthState.unauthenticated(),
+    );
+  }
+
+  void _onAuthLogout(AuthLogout event, Emitter<AuthState> emit) {
+    unawaited(_authenticationRepository.signOut());
```

Comment:
`"The method 'unawaited' isn't defined for the type 'AuthBloc'.\nTry correcting the name to the name of an existing method, or defining a method named 'unawaited'."`
I looked it up and found this: https://stackoverflow.com/a/70733424
What can we do about it?

Reply:
I think I can just remove it. I forgot that's for a future version of Dart.